File: homegenie_app/lib/core/models/service.dart
import 'package:json_annotation/json_annotation.dart';

part 'service.g.dart';

@JsonSerializable()
class Service {
  @JsonKey(name: 'id')
  final String id;

  @JsonKey(name: 'name')
  final String name;

  @JsonKey(name: 'description')
  final String description;

  @JsonKey(name: 'category')
  final String category;

  @JsonKey(name: 'basePrice')
  final double base_price;

  @JsonKey(name: 'durationHours')
  final double duration_hours;

  @JsonKey(name: 'isActive')
  final bool is_active;

  @JsonKey(name: 'requirements')
  final List<String> requirements;

  @JsonKey(name: 'includes')
  final List<String> includes;

  @JsonKey(name: 'excludes')
  final List<String> excludes;

  @JsonKey(name: 'imageUrl')
  final String? image_url;

  @JsonKey(name: 'createdAt')
  final DateTime created_at;

  @JsonKey(name: 'updatedAt')
  final DateTime updated_at;

  const Service({
    required this.id,
    required this.name,
    required this.description,
    required this.category,
    required this.base_price,
    required this.duration_hours,
    required this.is_active,
    required this.requirements,
    required this.includes,
    required this.excludes,
    this.image_url,
    required this.created_at,
    required this.updated_at,
  });

  factory Service.fromJson(Map<String, dynamic> json) => _$ServiceFromJson(json);
  Map<String, dynamic> toJson() => _$ServiceToJson(this);
}

File: shared/lib/src/types/service.ts
import { z } from 'npm:zod';
import { UuidSchema, TimestampSchema } from './base.ts';

// Service categories
export const ServiceCategorySchema = z.enum([
  'cleaning',
  'plumbing',
  'electrical',
  'gardening',
  'handyman',
  'beauty',
  'appliance_repair',
  'painting',
  'pest_control',
  'home_security'
]);

// Service schema
export const ServiceSchema = z.object({
  id: UuidSchema,
  name: z.string().min(2).max(100),
  description: z.string().max(500),
  category: ServiceCategorySchema,
  basePrice: z.number().min(0),
  durationHours: z.number().min(0.5).max(24),
  isActive: z.boolean().default(true),
  requirements: z.array(z.string()).default([]),
  includes: z.array(z.string()).default([]),
  excludes: z.array(z.string()).default([]),
  imageUrl: z.string().url().optional(),
  createdAt: TimestampSchema,
  updatedAt: TimestampSchema,
});

// Service pricing tiers
export const PricingTierSchema = z.object({
  id: UuidSchema,
  serviceId: UuidSchema,
  name: z.string(), // e.g., "1 BHK", "2 BHK", "3 BHK"
  description: z.string().optional(),
  price: z.number().min(0),
  durationHours: z.number().min(0.5),
  isDefault: z.boolean().default(false),
});

export type ServiceCategory = z.infer<typeof ServiceCategorySchema>;
export type Service = z.infer<typeof ServiceSchema>;
export type PricingTier = z.infer<typeof PricingTierSchema>;